tests/Types/SmallFloatTest.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Tests\Types;
+
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Types\SmallFloatType;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+class SmallFloatTest extends TestCase
+{
+    private AbstractPlatform&MockObject $platform;
+    private SmallFloatType $type;
+
+    protected function setUp(): void
+    {
+        $this->platform = $this->createMock(AbstractPlatform::class);
+        $this->type     = new SmallFloatType();
+    }
+
+    public function testFloatConvertsToPHPValue(): void
+    {
+        self::assertEquals(5.5, $this->type->convertToPHPValue('5.5', $this->platform));
+    }
+
+    public function testFloatNullConvertsToPHPValue(): void
+    {
+        self::assertNull($this->type->convertToPHPValue(null, $this->platform));
+    }
+
+    public function testFloatConvertToDatabaseValue(): void
+    {
+        self::assertIsFloat($this->type->convertToDatabaseValue(5.5, $this->platform));
```

Comment:
We've had this discussion already. Don't just accept _any_ float value here. This assertion is too weak.